### DISCUSSION TITLE
Using multiple accounts on macOS

### DISCUSSION THREAD

**Discussion by Abd-Ur-Rehman** [community/community#23829] [New to GitHub]
<p>Hi there! </p>
<p>I’m having trouble with the SSH keys while using multiple accounts on macOS the key I add 1st into SSH agent only works, the 2nd one doesn’t work. Thoughts? How can them both working at the same time or how to setup the accounts? </p>
<p>Thanks in advance!</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/abd-ur-rehman">@abd-ur-rehman</a>,</p>
<p>Thank you for being here! Working with multiple accounts and multiple ssh keys can be a tricky thing to set up.</p>
<p>Most of the time we recommend against having multiple accounts specifically for this reason, and usually just recommend that users use a single account for both work and personal projects. Still, we know this isn’t a good solution for everyone, so hopefully, we can point you in the right direction for setting this up!</p>
<p>To do this you’ll need to create an ssh config file in:</p>
<pre><code>~/.ssh/config
</code></pre>
<p>The contents should be similar to this, defining two different SSH hosts that point to GitHub but use different keys. You’ll need to modify the contents to point to your two separate SSH keys in the <code>IdentityFile</code> section:</p>
<pre><code>Host github.com
HostName github.com
PreferredAuthentications publickey
IdentityFile ~/.ssh/id_rsa_personal

Host work.github.com
HostName github.com
PreferredAuthentications publickey
IdentityFile ~/.ssh/id_rsa_work
</code></pre>
<p>This config sets it up so that every time you clone a repository using a standard SSH URL (<code>git@github.com:username/repo.git</code>) it will use your personal account key, but if you modify the URL slightly you can tell it to use your work key instead (<code>git@work.github.com:username/repo.git</code>).</p>
<p>I hope this helps!</p>

**Comment by Abd-Ur-Rehman** [comment, marked as answer]
<p>Thanks for your reply. I already have found a solution from <a href="https://code.tutsplus.com/tutorials/quick-tip-how-to-work-with-github-and-multiple-accounts--net-22574" rel="nofollow noopener">here</a></p>

**Comment by aadishsharechat** [comment]
<p>In My Case this worked</p>
<pre><code>Host github.com
IdentityFile ~/.ssh/id_rsa_personal

Host work.github.com
Host github.com
IdentityFile ~/.ssh/id_rsa_work
</code></pre>
